azure/types.go

```diff
@@ -135,6 +137,13 @@ func (vmss VMSS) InstancesByProviderID() map[string]VMSSVM {
 
 // ProviderID returns the K8s provider ID for the VMSS instance.
 func (vm VMSSVM) ProviderID() string {
+	if vm.InstanceID == "" {
+		// build providerID a bit differently for VMSS Flex
```

Comment:
What does "a bit differently" mean? this code is not very human friendly, a more explicit code comment with a concrete example could be helpful